lib/expressions.js

```diff
@@ -140,6 +206,28 @@ expressions.rotate = {
     var v = new THREE.Vector3(this.axis[0],this.axis[1],this.axis[2]);
     this.matrix.makeRotationAxis(v, this.angle);
   },
+  bounds: function() {
+    // find max distance to rotation axis, expand cube to that *sqrt(2) around axis.
+    // this could likely be optimized, but calc trade-off?
+    if (this.angle === 0) {
+      // trivial
+      return this.operands[0].bounds();
+    }
+    // TODO: introduce rotation center rotation? what if this.origin is not 0?
+    var origin = [0, 0, 0];
+    var distances = [];
+    for (var b = 0; b < 2; b++) {
+      for (var i = 0; i < 3; i++) {
+        distances.push(Math.abs(this.operands[0].bounds()[b][i] - origin[i]));
+      }
+    }
+    // factor sqrt(2) for Pi/4 turn of a cube
+    var maxd = Math.max.apply(this, distances) * Math.sqrt(2);
```

Comment:
Would be better to apply the inverse rotation to the sample point to see if it's in the bounds.  Probably that is the better approach for all the transforms.  I.e. the transforms are all resampling transforms, taking world space points into the local space of the expressions.

Now that I think about it bounds should be isInBounds should be a virtual method that returns true by default but can be implemented for most expression types.  Then an _bounds method can calculate and cache the bounds of the subexpressions for efficient evaluation.

I'm thinking of the cases where there are nonlinear transforms and/or procedural expressions where isInBounds is easy to calculate but bounds is not.  

For some complex modeling tasks being able to trivially reject large portions of the grid will be great for efficiency!